evm_arithmetization/src/cpu/kernel/asm/cdk_pre_execution.asm

```diff
@@ -0,0 +1,118 @@
+/// CDK-Erigon pre-block execution logic.
+/// Reference implementation: `cdk-erigon/core/state/intra_block_state_zkevm.go`.
+/// This currently supports the Etrog upgrade.
+
+/// Pre-stack: (empty)
+/// Post-stack: (empty)
+global pre_block_execution:
+    // stack: (empty)
+    PUSH txn_loop
+    // stack: retdest
+    PUSH @ADDRESS_SCALABLE_L2
+    %is_non_existent
+    %jumpi(create_scalable_l2_account)
+
+global update_scalable_block_number:
+    // stack: retdest
+    %blocknumber
+    PUSH @LAST_BLOCK_STORAGE_POS
+    // stack: last_block_slot, block_number, retdest
+    %write_scalable_storage
+    // stack: retdest
+
+    // Check timestamp
+    PUSH @ADDRESS_SCALABLE_L2_STATE_KEY
+    PUSH @TIMESTAMP_STORAGE_POS
+    %read_storage_linked_list_w_state_key
+    // stack: old_timestamp, retdest
+    %timestamp
+    GT %jumpi(update_scalable_timestamp)
+
+global update_scalable_prev_block_root_hash:
+    // stack: retdest
+    %mload_global_metadata(@GLOBAL_METADATA_STATE_TRIE_DIGEST_BEFORE)
+    // stack: prev_block_root, retdest
+    PUSH @STATE_ROOT_STORAGE_POS
+    PUSH 1 %blocknumber SUB
+    // stack: block_number - 1, STATE_ROOT_STORAGE_POS, prev_block_root, retdest
+    PUSH @SEGMENT_KERNEL_GENERAL
+    // stack: addr, block_number - 1, STATE_ROOT_STORAGE_POS, prev_block_root, retdest
+    MSTORE_32BYTES_32
+    // stack: addr, STATE_ROOT_STORAGE_POS, prev_block_root, retdest
+    MSTORE_32BYTES_32
+    // stack: addr, prev_block_root, retdest
+    POP
+    // stack: prev_block_root, retdest
+    PUSH 64 PUSH @SEGMENT_KERNEL_GENERAL
+    // stack: addr, len, prev_block_root, retdest
+    KECCAK_GENERAL
+    // stack: slot, prev_block_root, retdest
+    %write_scalable_storage
+    // stack: retdest
+
+global update_scalable_l1blockhash:
+    // stack: retdest
+    PROVER_INPUT(ger)
+    // stack: l1blockhash?, retdest
+    DUP1 %eq_const(@U256_MAX) %jumpi(skip_and_exit)
```

Comment:
From what I understand, `PROVER_INPUT` only returns `U256_MAX` if `ger_data` is None, right? So can this be the case when `cdk_erigon` is activated?

Reply:
Yes we _may_ not have anything to write, in which case we'll exit early

Reply:
`ger_data` will be None for non-cdk networks, but can also be `None` in case there's no update